src/test/mina_automation/executor.ml

```diff
@@ -11,83 +11,112 @@ module DockerContext = struct
     { image : string; workdir : string; volume : string; network : string }
 end
 
-type context =
-  | Dune (* application ran from dune exec command *)
-  | Local (* application ran from _build/default folder*)
-  | Debian (* application installed from mina debian package *)
-  | Docker of DockerContext.t
-  | AutoDetect
 (* application ran inside docker container *)
 
-module Executor = struct
-  type t = { official_name : string; dune_name : string; context : context }
+module type AppPaths = sig
+  (** The name of the application as it appears in the dune file. *)
+  val dune_name : string
 
-  let of_context ~context ~dune_name ~official_name =
-    { context; dune_name; official_name }
-
-  let run_from_debian t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "." t.official_name args
-
-  let run_from_dune t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "." "dune" ([ "exec"; t.dune_name; "--" ] @ args)
-
-  let run_from_local t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "."
-      (Printf.sprintf "_build/default/%s" t.dune_name)
-      args
+  (** The name of the application as it appears in the debian package or docker image. *)
+  val official_name : string
+end
 
-  let built_name t = Printf.sprintf "_build/default/%s" t.dune_name
+let logger = Logger.create ()
 
+module PathFinder = struct
```

Comment:
Yes, I agree the original implementation is nicer. We could revert to that version, just I still think we should avoid this field that cause a functor to boom on construction when possible. 

```ocaml
  let standalone_path_exn =
    Deferred.map standalone_path ~f:(fun opt ->
        Option.value_exn opt
          ~message:
            "Could not find standalone path. App is not executable outside the \
             dune" )
```

Reply:
Just remove this function and unwrap the option at callsites, then we should be fine. Right now the situation we'll panic depends on the scheduling of this Deferred, which is not great I think.